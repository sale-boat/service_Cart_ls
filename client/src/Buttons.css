.buttons {
    margin-top : 15px;
}

#quantity{
    width : 40px;
    border : 1px solid #DDD;
}

#quantity option{
  width : 40px;
}

.qty{
    font-size: 13px;
}

.cart{
    display: block;
    width : 100%;
    height: 30px;
    margin-top : 20px;

}
.addToCart{
    position: relative;
    display: block;
    width: 100%;
    height: 31px;
    background: linear-gradient(to bottom,#f7dfa5,#f0c14b);
    border-color: #a88734 #9c7e31 #846a29;
    padding: 8px;
    border-radius: 3px;
    font-size: 13px;
    text-align: middle;
}

.add{
    font-size: 13px;
    color: #111;
    vertical-align: middle;
}

.cartimage{
    top: 2px;
    left: 2px;
    position: absolute;
    height : 25px;
    width : 25px;
    vertical-align : middle;
}

.buyNow{
    display: block; 
    width: 100%;
    margin-top: 10px;
}

.buy {
    position: relative;
    display: block;
    width: 100%;
    height: 31px;
    background: linear-gradient(to bottom,#f6c88f,#ed9220);
    border-color: #ca7c1b #be751a #a56616;
    padding: 8px;
    border-radius: 3px;
    font-size: 13px;
    text-align: middle;
}

.buyer{
    font-size: 13px;
    color: #111;
    text-align: center;
}

.buyimage{
    top: 2px;
    left: 2px;
    position: absolute;
    height: 25px;
    width: 25px;
    vertical-align: middle;
}