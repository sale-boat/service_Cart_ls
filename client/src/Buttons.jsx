import React from 'react';
import button from './Buttons.css';
import axios from 'axios';

class Buttons extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      quantity: [],
    };
  }

  componentDidMount() {
    const { quantity } = this.state;
    const urlArray = document.URL.split('/');
    const productID = Number(urlArray[urlArray.length - 1]);
    axios.get(`/api/product/${productID}`)
      .then((res) => {
        for (let i = 1; i < res.data[0].quantity + 1; i += 1) {
          quantity.push(<option>{i}</option>);
        }
      })
      .catch((err) => {
        console.log(err);
      });
  }

  render() {
    const { quantity } = this.state;
    const style = { visibility: quantity ? 'visible' : 'hidden' };
    return (
      <div className={button.buttons} style={style}>
        <span className={button.qty}>
          {'Qty:'}
          {' '}
        </span>
        <select id={button.quantity}>
          { quantity }
        </select>
        <div className={button.cart}>
          <button className={button.addToCart} type="button">
            <img className={button.cartimage} src="https://s3.us-east-2.amazonaws.com/chrisfakephotos/amazoncart3.png" alt="cart" />
            {' Add To Cart'}
          </button>
        </div>
        <div className={button.buyNow}>
          <button className={button.buy} type="button">
            <img className={button.buyimage} src="https://s3.us-east-2.amazonaws.com/chrisfakephotos/amazonbuynow.png" alt="buynow" />
            {' Buy Now'}
          </button>
        </div>
      </div>
    );
  }
}


export default Buttons;
