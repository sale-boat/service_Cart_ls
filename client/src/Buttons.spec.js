import React from 'react';
import { shallow } from 'enzyme';
import Buttons from './Buttons.jsx';

describe('Buttons Component', () => {
  it('should render', () => {
    const buttons = shallow(<Buttons />);
    expect(buttons).toMatchSnapshot();
  });
});