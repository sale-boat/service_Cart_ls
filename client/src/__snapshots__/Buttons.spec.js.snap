// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Component should render 1`] = `
<div
  className="buttons"
>
  <span
    className="qty"
  >
    Qty:
     
  </span>
  <select
    id="quantity"
  />
  <div
    className="cart"
  >
    <button
      className="addToCart"
      type="button"
    >
      <img
        alt="cart"
        className="cartimage"
        src="https://s3.us-east-2.amazonaws.com/chrisfakephotos/amazoncart3.png"
      />
       Add To Cart
    </button>
  </div>
  <div
    className="buyNow"
  >
    <button
      className="buy"
      type="button"
    >
      <img
        alt="buynow"
        className="buyimage"
        src="https://s3.us-east-2.amazonaws.com/chrisfakephotos/amazonbuynow.png"
      />
       Buy Now
    </button>
  </div>
</div>
`;
